### PR TITLE
tiflash: Add note about mpp plan change

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

The default behavior of the generated plan to tiflash has been changed since v7.3.0, but we didn't mention it in the release note.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/44324, https://asktug.com/t/topic/1020202

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-7.3.0.md [hunk 1]
```diff
@@ -152,6 +152,10 @@ v7.3.0 引入了以下主要功能。[功能详情](#功能详情)中列出的
 
 ### 行为变更
 
+* TiDB
+
+    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。自此版本起，TiDB 给 TiFlash 下推计算任务时，默认只生成 MPP 协议。如果用户设置了 `tidb_allow_mpp=OFF`，在升级后查询可能会报错。建议用户在升级前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。
```
Comment (windtalker) [inline releases/release-7.3.0.md:157]: 再加上 如需强制使用 cop 协议，需要设置 `tidb_allow_tiflash_cop` 为 `ON`?

Comment (JaySon-Huang) [reply releases/release-7.3.0.md]: done

**Review by windtalker — APPROVED** [review]
LGTM

**Review by joey-yez — APPROVED** [review]

**Comment by JaySon-Huang** [comment]
@hfxsd PTAL

File: releases/release-7.3.0.md [hunk 2]
```diff
@@ -152,6 +152,10 @@ v7.3.0 引入了以下主要功能。[功能详情](#功能详情)中列出的
 
 ### 行为变更
 
+* TiDB
+
+    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。自此版本起，TiDB 给 TiFlash 下推计算任务时，默认只生成 MPP 协议。如果用户设置了 `tidb_allow_mpp=OFF`，在升级后查询可能会报错。建议用户在升级前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果需要生成 Cop、BatchCop 协议，可以将 `tidb_allow_tiflash_cop` 设置为 `ON`。
```
Comment (qiancai) [inline releases/release-7.3.0.md:157]:
```suggestion
    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。从 v7.3.0 起，当 TiDB 向 TiFlash 下推计算任务时，优化器默认只生成使用 MPP 协议的执行计划。如果设置了 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 为 `OFF`，在升级后查询可能会报错。建议在升级前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果需要优化器根据成本估算从 Cop、BatchCop 和 MPP 协议中选择一个用于生成执行计划，可以将 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量设置为 `ON`。
```

File: releases/release-7.3.0.md [hunk 3]
```diff
@@ -152,6 +152,10 @@ v7.3.0 引入了以下主要功能。[功能详情](#功能详情)中列出的
...
+
```
Comment (qiancai) [inline releases/release-7.3.0.md:158]: 是否需要在 tidb_allow_mpp 变量的描述里也加一句说明呢，比如在下面绿框的位置加一句类似这样的话 “如果将该变量值设置为 OFF，你需要同时开启 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量，否则可能遇到查询报错。”

image

File: releases/release-7.3.0.md [hunk 4]
```diff
@@ -152,6 +152,10 @@ v7.3.0 引入了以下主要功能。[功能详情](#功能详情)中列出的
 
 ### 行为变更
 
+* TiDB
+
+    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。从 v7.3.0 起，当 TiDB 向 TiFlash 下推计算任务时，优化器默认只生成使用 MPP 协议的执行计划。如果设置了 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 为 `OFF`，在升级后查询可能会报错。建议在升级 TiDB 前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果需要优化器根据成本估算从 Cop、BatchCop 和 MPP 协议中选择一个用于生成执行计划，可以将 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量设置为 `ON`。
```
Comment (JaySon-Huang) [inline releases/release-7.3.0.md:157]:
```suggestion
    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。从 v7.3.0 起，当 TiDB 向 TiFlash 下推计算任务时，优化器默认只生成使用 MPP 协议的执行计划。如果设置了 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 为 `OFF`，在升级后查询可能会报错。建议在升级 TiDB 前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果仍然需要优化器根据成本估算从 Cop、BatchCop 和 MPP 协议中选择一个用于生成执行计划，可以将 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量设置为 `ON`。
```

File: releases/release-7.3.0.md [hunk 5]
```diff
@@ -152,6 +152,10 @@ v7.3.0 引入了以下主要功能。[功能详情](#功能详情)中列出的
 
 ### 行为变更
 
+* TiDB
+
+    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。从 v7.3.0 起，当 TiDB 向 TiFlash 下推计算任务时，优化器默认只生成使用 MPP 协议的执行计划。如果设置了 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 为 `OFF`，在升级后查询可能会报错。建议在升级 TiDB 前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果仍然需要优化器根据成本估算从 Cop、BatchCop 和 MPP 协议中选择一个用于生成执行计划，可以将 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量设置为 `ON`。
```
Comment (JaySon-Huang) [inline releases/release-7.3.0.md:157]:
```suggestion
    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。从 v7.3.0 起，当 TiDB 向 TiFlash 下推计算任务时，优化器默认只生成使用 MPP 协议的执行计划。如果设置了 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 为 `OFF`，在升级后查询可能会报错，建议在升级 TiDB 前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果仍然需要优化器根据成本估算从 Cop、BatchCop 和 MPP 协议中选择一个用于生成执行计划，可以将 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量设置为 `ON`。
```

File: releases/release-7.3.0.md (hunk 3 above)
Comment (JaySon-Huang) [reply releases/release-7.3.0.md:158]: Done

File: releases/release-7.3.0.md [hunk 6]
```diff
@@ -152,6 +152,10 @@ v7.3.0 引入了以下主要功能。[功能详情](#功能详情)中列出的
 
 ### 行为变更
 
+* TiDB
+
+    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。从 v7.3.0 起，当 TiDB 向 TiFlash 下推计算任务时，优化器默认只生成使用 MPP 协议的执行计划。如果设置了 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 为 `OFF`，在升级后查询可能会报错，建议在升级 TiDB 前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果仍然需要优化器根据成本估算从 Cop、BatchCop 和 MPP 协议中选择一个用于生成执行计划，可以将 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量设置为 `ON`。
```
Comment (qiancai) [inline releases/release-7.3.0.md:157]:
```suggestion
    - MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。相对其他协议，MPP 协议更加成熟，能提供更好的任务和资源管理。从 v7.3.0 起，当 TiDB 向 TiFlash 下推计算任务时，优化器默认只生成使用 MPP 协议的执行计划。如果设置了 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 为 `OFF`，在升级 TiDB 后查询可能会报错，建议在升级前检查 `tidb_allow_mpp` 的值并将其设置为 `ON`。如果仍然需要优化器根据成本估算从 Cop、BatchCop 和 MPP 协议中选择一个用于生成执行计划，可以将 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量设置为 `ON`。
```

File: system-variables.md [hunk 7]
```diff
@@ -951,8 +951,8 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 类型：布尔型
 - 默认值：`ON`
 - 这个变量用于控制是否使用 TiFlash 的 MPP 模式执行查询，可以设置的值包括：
-    - 0 或 OFF，代表从不使用 MPP 模式
-    - 1 或 ON，代表由优化器根据代价估算选择是否使用 MPP 模式（默认）
+    - `0` 或 `OFF`，代表从不使用 MPP 模式。如果在 v7.3.0 及之后的版本将该变量值设置为 `OFF`，你需要同时开启 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量，否则可能遇到查询报错。
```
Comment (qiancai) [inline system-variables.md:954]:
```suggestion
    - `0` 或 `OFF`，代表从不使用 MPP 模式。如果在 v7.3.0 及之后的版本将该变量值设置为 `0` 或 `OFF`，你需要同时开启 [`tidb_allow_tiflash_cop`](/system-variables.md#tidb_allow_tiflash_cop-从-v730-版本开始引入) 变量，否则可能遇到查询报错。
```

File: system-variables.md [hunk 8]
```diff
@@ -951,8 +951,8 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
...
+    - `1` 或 `ON`，代表由优化器根据代价估算选择是否使用 MPP 模式（默认）
```
Comment (qiancai) [inline system-variables.md:955]:
```suggestion
    - `1` 或 `ON`，代表由优化器根据代价估算选择是否使用 MPP 模式（默认）。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**